packages/layer-composer/src/generators/vessel-events/vessel-events.ts

```diff
@@ -110,15 +120,44 @@ class VesselsEventsGenerator {
         metadata: {
           group: Group.Point,
         },
-      },
+      } as SymbolLayer,
+    ]
+    if (!config.showTrackSegments) {
```

Comment:
```suggestion
    if (!config.zoom <= POINTS_TO_SEGMENTS_ZOOM_LEVEL_SWITCH) {
```

Reply:
Good catch! I'll replace it with 
```js
(!config.track || (config.zoom as number) <= SEGMENTS_MIN_ZOOM_LEVEL)
```
this way we ensure we calculate the segments only when necessary and when generator has data